src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    public static final String DATE_FORMAT = "d.MM.yyyy";
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
+    public static final String DATA_SEPARATOR = " - ";
+    public static final String COLUMN_SEPARATOR = " ";
+    public static final String REPORT_HEADER = "Report for period ";
+    public static final int DATE_COLUMN_INDEX = 0;
+    public static final int NAME_COLUMN_INDEX = 1;
+    public static final int QOTY_COLUMN_INDEX = 2;
+    public static final int SUM_COLUMN_INDEX = 3;
+
+    private int getTotalSalaryByName(String[] data, String name,
+                                     LocalDate dateFromLocalDate, LocalDate dateToLocalDate) {
+        int totalSalary = 0;
+        for (String dataRow : data) {
+            if (dataRow.isEmpty()) {
+                continue;
+            }
+            String[] dataRowArray = dataRow.split(COLUMN_SEPARATOR);
+            LocalDate date;
+            try {
+                date = LocalDate.parse(dataRowArray[DATE_COLUMN_INDEX], formatter);
+            } catch (DateTimeParseException e) {
+                throw new DateTimeParseException("Error paring date in data rows",
+                        e.getParsedString(), e.getErrorIndex());
+            }
+            if ((date.isAfter(dateFromLocalDate) || date.equals(dateFromLocalDate))
+                    && (date.isBefore(dateToLocalDate) || date.equals(dateToLocalDate))
+                    && dataRowArray[NAME_COLUMN_INDEX].equals(name)) {
```

Comment:
```suggestion
            if ((date.isAfter(dateFromLocalDate) && (date.isBefore(dateToLocalDate)) || date.equals(dateToLocalDate))
                    && dataRowArray[NAME_COLUMN_INDEX].equals(name)) {
```
I guess  you can simplify this conditional